lib/net/ldap/connection.rb

```diff
@@ -652,4 +661,33 @@ def delete(args)
 
     pdu
   end
+
+  # Internal: Returns a Socket like object used internally to communicate with
+  # LDAP server.
+  #
+  # Typically a TCPSocket, but can be a OpenSSL::SSL::SSLSocket
+  def socket
+    return @conn if defined? @conn
```

Comment:
@satoryu This is likely done to avoid warnings from the interpreter for using uninitialized instance variables.  The `defined?` method is exempted from the warnings as I recall.